src/components/Navigation/NavBar.js

```diff
@@ -9,11 +9,38 @@ import Glyphicon from 'react-bootstrap/lib/Glyphicon';
 import Navbar from 'react-bootstrap/lib/Navbar';
 import FormGroup from 'react-bootstrap/lib/FormGroup';
 import FormControl from 'react-bootstrap/lib/FormControl';
-
+import MenuItem from 'react-bootstrap/lib/MenuItem';
+import Dropdown from 'react-bootstrap/lib/Dropdown';
 import NavLink from './NavLink';
-import FlagGroup from './FlagGroup';
 
 export const NavBar = React.createClass({
+
+  getInitialState() {
+    return {
+      selectedLanguage: 'Norsk'
+    };
+  },
+
+  handle(language) {
+    switch (language) {
+      case 'norway':
+        this.props.setLanguage(language);
+        this.setState({selectedLanguage: 'Norsk'});
+        break;
+      case 'norway-nynorsk':
+        this.props.setLanguage(language);
+        this.setState({selectedLanguage: 'Nynorsk'});
+        break;
+      case 'sweden':
+        this.props.setLanguage(language);
+        this.setState({selectedLanguage: 'Svensk'});
+        break;
+      case 'denmark':
+        this.props.setLanguage(language);
+        this.setState({selectedLanguage: 'Dansk'});
```

Comment:
Det er unødvendig å ha intern state i tillegg til state i redux-treet for `selectedLanguage`.
Hvis det er for å oversette språknavnet til norsk, så kan det gjøres i en egen funksjon:
```
getNorwegianLanguageName(language) {
  switch(language){
    case 'danish': return 'Dansk';
    case ...
  }
}
```